### DISCUSSION TITLE
Linking a created maintainer to the logged in user

### DISCUSSION THREAD

**Discussion by rberlin01** [irrdnet/irrd#887] [Q&A]
When logged in via HTTP GUI, and using the RPSL submission form to create a new maintainer, the maintainer does not link to the user that is signed in. I have scoured the documentation and have not figured out the proper way to do this. I was able to link my user manually to the maintainer via manual database entries, but I suspect this isn't the intention.

To add some context, I'm using IRRd for an internal project (non public internet), and I want every member of this internet to have their own maintainer and login to view/manage it. So far, all is working, except for this issue where the user doesn't automatically get related to the created maintainer, and I do not see a method for doing so.

Any insight would be awesome! Thank you for maintaining the project!

**Comment by fischerdouglas** [comment]
I think this could be related to [this ](https://github.com/irrdnet/irrd/issues/617#issuecomment-1026081054) suggestion I've made on #617.

> Have you considered to support Oauth integrations?
> PeeringDB already supports to be an OAuth base.
> 
> I believe we will see something in this direction by the RIRs in some time from now.
> And that would be fantastic to IRRs.

**Reply by rberlin01** [reply]
Thank you for the reply. I do see mention of migrated mntners automatically being linked to the authenticated user, but not for newly created mntners. Since, in my case, I'm trying to use it for a non-public project, there is no mntners to migrate so we're just creating them.

**Comment by mxsasha** [comment]
How are you creating these maintainers? Because they can [only be created by someone with override access](https://github.com/irrdnet/irrd/blob/stable/4.4.x/irrd/updates/validators.py#L392-L395), who then already has access to the maintainer. Not entirely following the process here.
